Raspberry Pi Survey 20250102

Cyanoxygen opened the pull request:
Topic Description
-----------------

- xorg-server: add config file for Raspberry Pi VideoCore GPU
    Raspberry Pi 5 exposes both of its V3D GPU and VideoCore VII display
    adapter. This confuses X11 as it does not know which card to choose.
    Adding this config file to let X11 know the correct card to work with.
    Reference: https://forums.raspberrypi.com/viewtopic.php?p=2159161#p2159161
- rpi-firmware-boot: rework the postinst script
    - Also detect chroot environments.
    - Detect the boot partition in fstab only if the partition is not
    mounted.
    - systemd-detect-virt can not be invoked with both -r and -c specified,
    and the behavior depends on the order of these two options. For now
    just run the checks separately.
- linux+kernel+rpi64+lts: update to 6.6.69
    - Add firmware packages as dependency.
- linux-kernel-rpi64-lts: update to 6.6.69
    - Enable more console fonts (to allow console texts on HiDPI displays to
    be readable).
    - Integate more filesystem drivers into the kernel.
    - Enable kernel image compression (Gzip).
    - Also check for chroot environment.

Package(s) Affected
-------------------

- linux+kernel+rpi64+lts: 6.6.69
- linux-kernel-rpi64-lts-6.6.69: 6.6.69
- rpi-firmware-boot: 1.20241126-1
- xorg-server: 21.1.15-1

Security Update?
----------------

No

Build Order
-----------

```
#buildit linux-kernel-rpi64-lts rpi-firmware-boot linux+kernel+rpi64+lts xorg-server
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AArch64 `arm64`

Cyanoxygen committed:
linux-kernel-rpi64-lts: update to 6.6.69

- Enable more console fonts (to allow console texts on HiDPI displays to
  be readable).
- Integate more filesystem drivers into the kernel.
- Enable kernel image compression (Gzip).
- Also check for chroot environment.

Cyanoxygen committed:
linux+kernel+rpi64+lts: update to 6.6.69

- Add firmware packages as dependency.

Cyanoxygen committed:
rpi-firmware-boot: rework the postinst script

- Also detect chroot environments.
- Detect the boot partition in fstab only if the partition is not
  mounted.
- systemd-detect-virt can not be invoked with both -r and -c specified,
  and the behavior depends on the order of these two options. For now
  just run the checks separately.

stdmnpkg commented:
works on my rpi3

Dustymind commented:
~~Package `rpi-firmware-boot` seems broken. Only `config.txt` and `distcfg.txt` were packaged.~~
~~Also, if I use [mkrawimg](https://github.com/AOSC-Dev/mkrawimg) to build image with this topic, only `config.txt` and `distcfg.txt` were copied to the boot partition.~~

~~By the way, the build log of `rpi-firmware-boot` seems odd.~~

EDIT: fixed in 7002a0a

Cyanoxygen commented:
Testing
-------

Flashed an image generated by mkrawimg with this topic. The image is bootable and usable.

![telegram-cloud-photo-size-1-4911472056084836054-y](https://github.com/user-attachments/assets/f24fb475-adc9-4b83-a4d1-f0f142ea97f8)

MingcongBai commented on runtime-display/xorg-config-rpi-vc4/autobuild/defines:
Consider merging this into xorg-server - there is a quirks system for matching drivers. At the very least, you can still ship this configuration with `xorg-server`.

Cyanoxygen replied:
Alright, merging into `xorg-server`.

Cyanoxygen committed:
xorg-server: add config file for Raspberry Pi VideoCore GPU

Raspberry Pi 5 exposes both of its V3D GPU and VideoCore VII display
adapter. This confuses X11 as it does not know which card to choose.
Adding this config file to let X11 know the correct card to work with.

Reference: https://forums.raspberrypi.com/viewtopic.php?p=2159161#p2159161

Dustymind commented on runtime-kernel/linux-kernel-rpi64-lts/autobuild/defines:
```diff
@@ -1,14 +1,13 @@
 PKGDEP=""
 BUILDDEP="bc dracut git pahole"
-PKGDES="Raspberry Pi downstream Linux Kernel v6.6.62 for AOSC OS (Longterm Maintenance)"
+PKGDES="Raspberry Pi downstream Linux Kernel v6.6.67 for AOSC OS (Longterm Maintenance)"
```
It should be `v6.6.69`, not `v6.6.67`.

Dustymind commented on runtime-kernel/linux-kernel-rpi64-lts/autobuild/postinst.in:
```diff
@@ -23,6 +23,8 @@ fi
 
 if systemd-detect-virt -q -c ; then
   echo "Running in build environment, skipping kernel update."
+elif systemd-detect-virt -q -r ; then
```
Since postinst detects both container and chroot environment, kernel and firmware update won't be triggered in the ***manual*** installation process, which was done in chroot environment (using `arch-chroot`).

If users want to use the ***manual*** installation, maybe we should inform them to manual update/install the kernel and firmware, both in the script and the wiki.

Cyanoxygen replied:
Yes, you just reminded me of that situation - removing the chroot check.

Dustymind commented on runtime-kernel/rpi-firmware-boot/autobuild/postinst:
```diff
@@ -1,46 +1,60 @@
+# WARNING! systemd-detect-virt can not be invoked with both -r and -c
+# specified. And the behavior depends on the order of these two options.
+# For now just run the checks separately.
+if systemd-detect-virt -q -r ; then
```
Ditto.

Dustymind commented:
LGTM on RPi4
![IMG_20250117_230716](https://github.com/user-attachments/assets/3a251861-a7ac-404b-bf41-9ae0a7c970dd)
